x-pack/plugins/security_solution/public/timelines/components/timeline/properties/helpers.tsx

```diff
@@ -104,55 +130,153 @@ interface DescriptionProps {
-  ({ description, timelineId, updateDescription }) => (
-    <EuiToolTip data-test-subj="timeline-description-tool-tip" content={i18n.DESCRIPTION_TOOL_TIP}>
-      <DescriptionContainer data-test-subj="description-container">
-        <EuiFieldText
-          aria-label={i18n.TIMELINE_DESCRIPTION}
-          data-test-subj="timeline-description"
-          fullWidth={true}
-          onChange={(e) => updateDescription({ id: timelineId, description: e.target.value })}
-          placeholder={i18n.DESCRIPTION}
-          spellCheck={true}
-          value={description}
-        />
+  ({
+    description,
+    timelineId,
+    updateDescription,
+    isTextArea = false,
+    disableAutoSave = false,
+    disableTooltip = false,
+    disabled = false,
+    marginRight,
+  }) => {
+    const onDescriptionChanged = useCallback(
+      (e) => {
+        updateDescription({ id: timelineId, description: e.target.value, disableAutoSave });
+      },
+      [updateDescription, disableAutoSave, timelineId]
+    );
+
+    const inputField = useMemo(
+      () =>
+        isTextArea ? (
+          <EuiTextArea
+            data-test-subj="timeline-description-textarea"
+            aria-label={i18n.TIMELINE_DESCRIPTION}
+            fullWidth={true}
+            onChange={onDescriptionChanged}
+            placeholder={i18n.DESCRIPTION}
+            value={description}
+            disabled={disabled}
+          />
+        ) : (
+          <EuiFieldText
+            aria-label={i18n.TIMELINE_DESCRIPTION}
+            data-test-subj="timeline-description"
+            fullWidth={true}
+            onChange={onDescriptionChanged}
+            placeholder={i18n.DESCRIPTION}
+            spellCheck={true}
+            value={description}
+          />
+        ),
+      [description, isTextArea, onDescriptionChanged, disabled]
+    );
+    return (
+      <DescriptionContainer data-test-subj="description-container" marginRight={marginRight}>
+        {disableTooltip ? (
+          inputField
+        ) : (
+          <EuiToolTip
+            data-test-subj="timeline-description-tool-tip"
+            content={i18n.DESCRIPTION_TOOL_TIP}
+          >
+            {inputField}
+          </EuiToolTip>
+        )}
       </DescriptionContainer>
-    </EuiToolTip>
-  )
+    );
+  }
 );
 Description.displayName = 'Description';
 
 interface NameProps {
+  autoFocus?: boolean;
+  className?: string;
+  disableAutoSave?: boolean;
+  disableTooltip?: boolean;
+  disabled?: boolean;
   timelineId: string;
   timelineType: TimelineType;
   title: string;
   updateTitle: UpdateTitle;
+  width?: string;
+  marginRight?: number;
 }
 
-export const Name = React.memo<NameProps>(({ timelineId, timelineType, title, updateTitle }) => {
-  const handleChange = useCallback((e) => updateTitle({ id: timelineId, title: e.target.value }), [
+export const Name = React.memo<NameProps>(
+  ({
+    autoFocus = false,
+    className = TIMELINE_TITLE_CLASSNAME,
```

Comment:
why do we need this `className`?

Reply:
yeah, I'm using that to auto focus the input field.

Reply:
@patrykkopycinski you're right! useRef works well for autoFocus, I've removed classname accordingly. Thank you 👍